### ISSUE TITLE
[2.9.0 regression] clearing a non-existing timeout id affects upcoming setTimeout

### ISSUE THREAD

**Issue by AviVahl** [jasmine/jasmine#1482]
https://github.com/jasmine/jasmine/pull/1427 introduced a regression.

After using `jasmine.clock().install()` in a *browser*, if the upcoming timeout id is cleared before setTimeout is called, tick() won't execute the callback:

## Expected Behavior
clearTimeout shouldn't stop upcoming timers from executing.

## Current Behavior
timeout callback never runs

## Possible Solution
https://github.com/jasmine/jasmine/blob/master/src/core/DelayedFunctionScheduler.js#L55
Move `deletedKeys.push(timeoutKey);` to the first line of the `if (i > -1) {`

## Suite that reproduces the behavior (for bugs)
_this would only break in web browsers. not in node. see context below..._
```javascript
jasmine.clock().install();
describe("sample", function() {
  it(() => {
    let val =1
    clearTimeout(0)
    setTimeout(() => val = 2, 2000) // asuming this is the first setTimeout since install() was called.
    jasmine.clock().tick(2000)
    expect(val).toBe(2)
  })
});
```
## Context
Real project. Real browser tests that pass (and should pass) with jasmine 2.8.0, but don't with 2.9.0
This issue can only happen in web browsers, where timeout ID is a primitive number. Node.js uses objects, so it's not affected.

## Your Environment
* Version used: 2.9.0
* Environment name and version: Chrome 63
* Operating System and version (desktop or mobile): win 10 x64

**Comment by slackersoft** [comment]
I haven't been able to reproduce this error, even when I copy your full example suite, and add a description for the `it` so the HtmlReporter displays results correctly.

I've also tried putting the `clock().install()` into a `beforeEach` where it expects to be (including the `clock().uninstall()` in an `afterEach`). Even so, the value is still updated correctly.

I tried to write a unit test for the DelayedFunctionScheduler even specifying the timeoutKey to use when scheduling, and it doesn't look like this is happening if the `clearTimeout` is called outside a `tick`.

Can you provide an example suite that actually fails in your browser so we can help fix the issue?

Hope this helps. Thanks for using Jasmine!

**Comment by AviVahl** [comment]
Sorry for the naive example. The original code that fails is closed source and involves a lot more code, as you can imagine. I'll try reproducing it in a clean environment and report back.

**Comment by AviVahl** [comment]
I was able to reproduce it with `clearTimeout(1)` not sure why I thought the first id was 0.
Steps to reproduce:
```
$ npm i create-react-app -g
$ create-react-app jasmine-bug
$ cd jasmine-bug
$ yarn add jasmine-core script-loader --dev
```
Then, change src/index.js to:
```
/* eslint-disable */

import 'jasmine-core/lib/jasmine-core/jasmine.css';
import '!script-loader!jasmine-core/lib/jasmine-core/jasmine';
import '!script-loader!jasmine-core/lib/jasmine-core/jasmine-html';
import '!script-loader!jasmine-core/lib/jasmine-core/boot';

describe('clearTimeout bug', () => {
    beforeEach(() => jasmine.clock().install())
    afterEach(() => jasmine.clock().uninstall())

    it('works', () => {
        let val = 1
        clearTimeout(1)
        setTimeout(() => val = 2, 2000) // asuming this is the first setTimeout since install() was called.
        jasmine.clock().tick(2000)
        expect(val).toBe(2)
    })
})
```
And run `yarn start`.

**Comment by slackersoft** [comment]
Thanks for the updated example. The fix should be out in 2.9.1 now.

**Comment by AviVahl** [comment]
Thanks for the prompt response and release! Much appreciated.
